sqldelight-idea-plugin/src/main/kotlin/com/squareup/sqldelight/intellij/actions/file/SqlDelightCreateFileAction.kt

```diff
@@ -0,0 +1,186 @@
+    return propsFile.databases.first().compilationUnits
+        .flatMap { it.sourceFolders }
+        .any { file.path.contains(it.path) }
+  }
+
+  override fun startInWriteAction() = false
+
+  override fun createFileFromTemplate(name: String, template: FileTemplate, dir: PsiDirectory) =
+      Companion.createFileFromTemplate(name, template, dir)
+
+  companion object {
+    private const val CAPTION = "SqlDelight File/Table"
+
+    private val iconProvider = SqlDelightFileIconProvider()
+    private val dummyFile = LightVirtualFile().apply { language = SqlDelightLanguage }
+
+    private val FQNAME_SEPARATORS = charArrayOf('/', '\\', '.')
+
+    private object NameValidator : InputValidatorEx {
+      override fun getErrorText(inputString: String): String? {
+        if (inputString.trim().isEmpty()) {
+          return "Name can't be empty"
+        }
+
+        val parts: List<String> = inputString.split(*FQNAME_SEPARATORS)
+        if (parts.any { it.trim().isEmpty() }) {
+          return "Name can't have empty parts"
+        }
+
+        return null
+      }
+
+      override fun checkInput(inputString: String): Boolean = true
+
+      override fun canClose(inputString: String): Boolean = getErrorText(inputString) == null
+    }
+
+    private fun findOrCreateTarget(dir: PsiDirectory, name: String, extension: String): Pair<String, PsiDirectory> {
+      var className = name.removeSuffix(".$extension")
+      var targetDir = dir
+
+      for (splitChar in FQNAME_SEPARATORS) {
+        if (splitChar in className) {
+          val names = className.trim().split(splitChar)
+
+          for (dirName in names.dropLast(1)) {
+            targetDir = targetDir.findSubdirectory(dirName) ?: runWriteAction {
+              targetDir.createSubdirectory(dirName)
+            }
+          }
+
+          className = names.last()
+          break
+        }
+      }
+      return className to targetDir
+    }
+
+    private fun createFromTemplate(dir: PsiDirectory, className: String, template: FileTemplate): PsiFile? {
+      val project = dir.project
+      val defaultProperties = FileTemplateManager.getInstance(project).defaultProperties
+
+      val properties = Properties(defaultProperties)
+
+      val element = try {
+        CreateFromTemplateDialog(
+            project, dir, template,
+            AttributesDefaults(className).withFixedName(true),
+            properties
+        ).create()
+      } catch (e: IncorrectOperationException) {
+        throw e
+      } catch (e: Exception) {
+        LOG.error(e)
+        return null
+      }
+
+      return element?.containingFile
+    }
+
+    private fun createFileFromTemplate(name: String, template: FileTemplate, dir: PsiDirectory): PsiFile? {
+      val (className, targetDir) = findOrCreateTarget(dir, name, template.extension)
+
+      val service = DumbService.getInstance(dir.project)
```

Comment:
were these just copied over? curious if not having it here causes any issues

Reply:
Yep they were, but I'll check, I also did not really understand the purpose of those